src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { ArtemisDataTableModule } from 'app/shared/data-table/data-table.module';
+import { AlertComponent } from 'app/shared/alert/alert.component';
+import { ArtemisDurationFromSecondsPipe } from 'app/shared/pipes/artemis-duration-from-seconds.pipe';
+import { ArtemisDatePipe } from 'app/shared/pipes/artemis-date.pipe';
+import { MockComponent, MockDirective, MockPipe, MockProvider } from 'ng-mocks';
+import { StudentExamService } from 'app/exam/manage/student-exams/student-exam.service';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { JhiAlertService, JhiTranslateDirective } from 'ng-jhipster';
+import { HttpResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+import { ActivatedRoute, convertToParamMap, Params } from '@angular/router';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+import { NgxDatatableModule } from '@swimlane/ngx-datatable';
+import { FontAwesomeTestingModule } from '@fortawesome/angular-fontawesome/testing';
+import { TranslateModule } from '@ngx-translate/core';
+import { RouterTestingModule } from '@angular/router/testing';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import * as chai from 'chai';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MockTranslateValuesDirective } from '../../../course/course-scores/course-scores.component.spec';
+import { Exercise } from 'app/entities/exercise.model';
+import { ModelingExercise, UMLDiagramType } from 'app/entities/modeling-exercise.model';
+import { ExerciseGroup } from 'app/entities/exercise-group.model';
+import { Exam } from 'app/entities/exam.model';
+import * as moment from 'moment';
+import { StudentParticipation } from 'app/entities/participation/student-participation.model';
+import { ParticipationType } from 'app/entities/participation/participation.model';
+import { Result } from 'app/entities/result.model';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { QuizExercise } from 'app/entities/quiz/quiz-exercise.model';
+import { FileUploadExercise } from 'app/entities/file-upload-exercise.model';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('StudentExamDetailComponent', () => {
+    let studentExamDetailComponentFixture: ComponentFixture<StudentExamDetailComponent>;
+    let studentExamDetailComponent: StudentExamDetailComponent;
+    let course: Course;
+    let student: User;
+    let studentExam: StudentExam;
+    let exercise: Exercise;
+    let exerciseGroup: ExerciseGroup;
+    let exam: Exam;
+    let studentParticipation: StudentParticipation;
+    let result: Result;
+
+    beforeEach(() => {
+        course = new Course();
+        course.id = 1;
+
+        student = new User();
+        student.name = 'name';
+        student.login = 'login';
+        student.email = 'email';
+        student.visibleRegistrationNumber = 'visibleRegistrationNumber';
```

Comment:
Same as above, you can save some extra lines by doing:
```
student: Student = { name: "name", login:"login", ... }
```

Reply:
Done